poptorch_experimental_addons/_impl/misc.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2023 Graphcore Ltd. All rights reserved.
+
+"""
+Small standalone utilities that don't belong in larger groups.
+"""
+
+from typing import Any, Tuple
+
+import poptorch
+import torch
+from torch import Tensor
+
+
+class _CustomGrad(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx: Any, fwd: Tensor, bwd: Tensor) -> Tensor:  # type:ignore[override]
+        return fwd
+
+    @staticmethod
+    def backward(  # type:ignore[override]
+        ctx: Any, grad: Tensor
+    ) -> Tuple[None, Tensor]:
+        return None, grad
+
+
+def custom_grad(fwd: Tensor, bwd: Tensor) -> Tensor:
```

Comment:
Would `bwd` be better off named as `surrogate` or `proxy` since you are not actually using the function defined here in the backward pass. Instead you are passing a surrogate function to differentiate instead. 

Term borrowed from popular usage in spiking neural network research e.g. https://arxiv.org/abs/1901.09948

Reply:
Ah, nice - thanks. I was struggling to name this one.